evaluator.plandefinition/src/main/java/org/opencds/cqf/cql/evaluator/plandefinition/r4/PlanDefinitionProcessor.java

```diff
@@ -0,0 +1,725 @@
+          "The planDefinition passed to FhirDal was " + "not a valid instance of PlanDefinition.class");
+    }
+
+    planDefinition = (PlanDefinition) basePlanDefinition;
+
+    logger.info("Performing $apply operation on PlanDefinition/" + theId);
+
+    CarePlan builder = new CarePlan();
+
+    builder.addInstantiatesCanonical(planDefinition.getIdElement().getIdPart()).setSubject(new Reference(patientId))
+        .setStatus(CarePlan.CarePlanStatus.DRAFT);
+
+    if (encounterId != null)
+      builder.setEncounter(new Reference(encounterId));
+    if (practitionerId != null)
+      builder.setAuthor(new Reference(practitionerId));
+    if (organizationId != null)
+      builder.setAuthor(new Reference(organizationId));
+    if (userLanguage != null)
+      builder.setLanguage(userLanguage);
+
+    // Each Group of actions shares a RequestGroup
+    RequestGroup requestGroup = new RequestGroup().setStatus(RequestStatus.DRAFT).setIntent(RequestIntent.PROPOSAL);
+
+    IBaseDatatype basePrefetchDataKey = operationParametersParser.getValueChild(prefetchData, "key");
+    String prefetchDataKey = null;
+    if (basePrefetchDataKey != null) {
+      if (!(basePrefetchDataKey instanceof StringType)) {
+        throw new IllegalArgumentException("prefetchData key must be a String");
+      }
+      prefetchDataKey = ((StringType) basePrefetchDataKey).asStringValue();
+    }
+
+    IBaseResource basePrefetchDataDescription = operationParametersParser.getResourceChild(prefetchData, "descriptor");
+    DataRequirement prefetchDataDescription = null;
+    if (basePrefetchDataDescription != null) {
+      if (!(basePrefetchDataDescription instanceof DataRequirement)) {
+        throw new IllegalArgumentException("prefetchData descriptor must be a DataRequirement");
+      }
+      prefetchDataDescription = ((DataRequirement) basePrefetchDataDescription);
+    }
+
+    IBaseResource basePrefetchDataData = operationParametersParser.getResourceChild(prefetchData, "data");
+    IBaseBundle prefetchDataData = null;
+    if (basePrefetchDataData != null) {
+      if (!(basePrefetchDataData instanceof IBaseBundle)) {
+        throw new IllegalArgumentException("prefetchData data must be a Bundle");
+      }
+      prefetchDataData = ((IBaseBundle) basePrefetchDataData);
+    }
+
+    Session session = new Session(planDefinition, builder, patientId, encounterId, practitionerId, organizationId,
+        userType, userLanguage, userTaskContext, setting, settingContext, requestGroup, parameters, prefetchData,
+        contentEndpoint, terminologyEndpoint, dataEndpoint, bundle, useServerData, mergeNestedCarePlans,
+        prefetchDataData, prefetchDataDescription, prefetchDataKey);
+
+    CarePlan carePlan = (CarePlan) ContainedHelper.liftContainedResourcesToParent(resolveActions(session));
+    IBaseParameters returnParameters = new Parameters();
+    operationParametersParser.addResourceChild(returnParameters, "return", carePlan);
+    return returnParameters;
```

Comment:
This should just return the CarePlan.

Reply:
hum... can't the apply also return a Task? Does it have to be a CarePlan?